images.go

```diff
@@ -167,6 +179,31 @@ func (r *imageStore) datapath(id, key string) string {
 	return filepath.Join(r.datadir(id), makeBigDataBaseName(key))
 }
 
+// bigDataNameIsManifest determines if a big data item with the specified name
+// is considered to be representative of the image, in that its digest can be
+// said to also be the image's digest.  Currently, if its name is, or begins
+// with, "manifest", we say that it is.
+func bigDataNameIsManifest(name string) bool {
+	return name == ImageDigestBigDataKey || strings.HasPrefix(name, ImageDigestBigDataKey)
+}
+
+// recomputeDigests takes the list of an image's digests and deduplicates them.
+func recomputeDigests(inputDigests []digest.Digest) (validDigests []digest.Digest, err error) {
```

Comment:
(`s/recompute/deduplicate/`, then? or `validateAndDeduplicate`, or… I don’t know; it’s just that this is definitely not recomputing the values.

Or maybe move more code from both callers, setting `.Digests` based on `.Digest` and `BigDataDigests`, in here.)

Reply:
Oh, that makes sense.  Moving some of that duplicated logic into `recomputeDigests()`.